bigframes/dtypes.py

```diff
@@ -578,6 +580,8 @@ def _infer_dtype_from_python_type(type: type) -> Dtype:
         return DATE_DTYPE
     if issubclass(type, datetime.time):
         return TIME_DTYPE
+    if issubclass(type, datetime.timedelta):
```

Comment:
Likewise, do we need numpy, pandas, and pyarrow object detection here too?

Reply:
Done. Left the PyArrow out for this one.